contracts/dca/src/handlers/execute.rs

```diff
@@ -240,52 +255,45 @@ fn cancel_dca(deps: DepsMut, info: MessageInfo, app: DCAApp, dca_id: String) ->
 
     DCA_LIST.remove(deps.storage, dca_id.clone());
 
-    // #3
-    let cron_cat = app.cron_cat(deps.as_ref());
-    let remove_task_msg = cron_cat.remove_task(dca_id)?;
+    // Remove task from Cron Cat
+    // #3.3
+    // Hint: Using the Cron Cat API
+    let remove_task_msg = CosmosMsg::Custom(cosmwasm_std::Empty {});
 
     Ok(app.tag_response(Response::new().add_message(remove_task_msg), "cancel_dca"))
 }
 
 /// Execute swap if called my croncat manager
 /// Refill task if needed
-fn convert(deps: DepsMut, env: Env, info: MessageInfo, app: DCAApp, dca_id: String) -> AppResult {
+fn convert(deps: DepsMut, _env: Env, info: MessageInfo, app: DCAApp, dca_id: String) -> AppResult {
     let config = CONFIG.load(deps.storage)?;
-    let dca = DCA_LIST.load(deps.storage, dca_id.clone())?;
-
-    let cron_cat = app.cron_cat(deps.as_ref());
+    let _dca = DCA_LIST.load(deps.storage, dca_id.clone())?;
 
-    let manager_addr = cron_cat.query_manager_addr(env.contract.address.clone(), dca_id.clone())?;
+    // Check if this method called by Cron Cat Manager
+    // #4
+    // Hint: Cron Cat API has method to get manager addr
+    let manager_addr = cosmwasm_std::Addr::unchecked("");
     if manager_addr != info.sender {
         return Err(AppError::NotManagerConvert {});
     }
     let mut messages = vec![];
 
     // In case task running out of balance - refill it
-    let task_balance = cron_cat
-        .query_task_balance(env.contract.address, dca_id.clone())?
-        .balance
-        .unwrap();
-    if task_balance.native_balance < config.refill_threshold {
-        messages.push(
-            cron_cat.refill_task(
-                dca_id,
-                AssetList::from(vec![Asset::native(
-                    config.native_denom,
-                    config.dca_creation_amount,
-                )])
-                .into(),
-            )?,
-        );
+    // #5
+    // If task balance is below config threshold - refill it
+    // Hint: You can get task balance and refill using Cron Cat API :)
+    // Don't be shy to unwrap here, since this method is called 
+    // by Cron Cat of this DCA task and it is not possible that there is no balance!
+    let task_balance = Uint128::new(1);
+    if task_balance < config.refill_threshold {
+        let refill_task_msg = CosmosMsg::Custom(cosmwasm_std::Empty {});
+        messages.push(refill_task_msg)
     }
 
-    // TODO: remove dca on failed swap?
-    // Or `stop_on_fail` should be enough
-    messages.push(app.dex(deps.as_ref(), dca.dex).swap(
-        dca.source_asset,
-        dca.target_asset,
-        Some(config.max_spread),
-        None,
-    )?);
+    // Finally do the swap!
+    // #6
+    // Hint: Using Dex API
+    // After you done - remove allow(unused) parts and underscores to make sure you used everything!
```

Comment:
@CyberHoward Or this?

Reply:
ahhh yeah sure

they can also run the tests and see if it works!

Reply:
Or single allow_unused at the start will be easier to manage nvm